packages/app/auth/user.ts

```diff
@@ -1,51 +1,12 @@
-import { getCookies } from 'cookies-next'
-import type { IncomingMessage, ServerResponse } from 'http'
-import type { NextApiRequest, NextApiResponse } from 'next'
-
-const LEGACY_API_HOST = process.env.API_HOST || 'meditor_legacy-api:8081'
-const LEGACY_MEDITOR_SESSION_COOKIE = '__mEditor'
+import { getServerSession as getServerSessionNextAuth } from 'next-auth'
+import { authOptions } from 'pages/api/auth/[...nextauth]'
 
 /**
- * for the time being, authentication is ONLY handled in the legacy API (/legacy-api)
+ * a wrapper over the NextAuth getServerSession
  *
- * In the future, we'll want to use NextAuth (without breaking .netrc based logins). This function matches the parameter definition
- * of the "unstable_getServerSideSession" function in NextAuth and can be augmented.
- */
-export async function getLoggedInUser(
-    req: NextApiRequest | IncomingMessage,
-    res: NextApiResponse | ServerResponse
-) {
-    // smooths out the difference between Next's request / response type and the native HTTP versions
-    const cookies = getCookies({ req, res })
-
-    return getUserBySessionCookie(cookies)
-}
-
-/**
- * if the user has a mEditor session cookie, we'll use the legacy API to get the logged in user's information
+ *? This doesn't seem necessary but the API has changed multiple times and is a bit verbose to use as every page would need to import
+ *? both getServerSession() and authOptions. This simplifies it a bit and abstracts away NextAuth from being used throughout the app
  */
-export async function getUserBySessionCookie(cookies: { [key: string]: string }) {
-    const legacySession = cookies[LEGACY_MEDITOR_SESSION_COOKIE]
-
-    if (!legacySession) {
-        // this user is not logged in using a legacy session cookie
-        return
-    }
-
-    const getMeResponse = await fetch(`http://${LEGACY_API_HOST}/meditor/api/me`, {
-        credentials: 'include',
-        headers: { Cookie: `${LEGACY_MEDITOR_SESSION_COOKIE}=${legacySession}` },
-    })
-
-    if (!getMeResponse.ok) {
-        // no legacy user found
-        return
-    }
-
-    const user = await getMeResponse.json()
-
-    return {
-        ...user,
-        name: [user.firstName, user.lastName].join(' '),
-    }
+export async function getServerSession(req: any, res: any) {
+    return getServerSessionNextAuth(req, res, authOptions)
```

Comment:
This file shrunk drastically. It's not even necessarily needed, but see the comment for more info why I wrapped NextAuth's getServerSession